Provider/PhpcrMenuProvider.php

```diff
@@ -157,116 +158,81 @@ public function setRequest(Request $request = null)
      * menu root. You can thus pass a name or any relative path with slashes to
      * only load a submenu rather than a whole menu.
      *
-     * @param string $name    Name of the menu to load. This can be an
-     *                        absolute PHPCR path or one relative to the menu root.
-     * @param array  $options
-     *
-     * @return ItemInterface The menu (sub)tree starting with name.
-     *
-     * @throws \InvalidArgumentException if the menu can not be found.
+     * {@inheritDoc}
      */
     public function get($name, array $options = array())
     {
-        $menu = $this->find($name, $options, true);
+        $document = $this->find($name, $options, true);
+
+        if (null === $document) {
+            throw new \InvalidArgumentException(sprintf(
+                'Menu "%s" could not be located by the PhpcrMenuProvider',
+                $name
+            ));
+        }
+
+        $menuItem = $this->loader->load($document);
 
-        $menuItem = $this->loader->load($menu);
         if (empty($menuItem)) {
-            throw new \InvalidArgumentException("Menu at '$name' is misconfigured (f.e. the route might be incorrect) and could therefore not be instanciated");
+            throw new \InvalidArgumentException(sprintf(
+                'Menu "%s" is misconfigured (f.e. the route might be incorrect) and could therefore not be instansiated',
+                $name
+            ));
         }
 
         return $menuItem;
     }
 
     /**
-     * Check if a menu node exists.
-     *
-     * If this method returns true, it means that you can call get() without
-     * an exception.
-     *
-     * @param string $name    Name of the menu to load. This can be an
-     *                        absolute PHPCR path or one relative to the menu root.
-     * @param array  $options
-     *
-     * @return bool Whether a menu with this name can be loaded by this provider.
+     * {@inheritDoc}
      */
     public function has($name, array $options = array())
     {
-        return $this->find($name, $options, false) instanceof NodeInterface;
+        $document = $this->find($name, $options);
+
+        if (null === $document) {
+            return false;
+        }
+
+        return true;
```

Comment:
this code is equivalent to `return $this->find($name, $options) !== null;`

i wonder if we should not do the instanceof check here and throw an exception if the interface is missing. its the earliest possible moment to detect the problem, which is usually the right place. the chain provider does not call has if a previous provider has the menu. and basically we are lying when we say we have the menu but when you fetch it we don't.